src/schema/schema_project.c

```diff
@@ -54,16 +54,18 @@ __wt_schema_project_in(WT_SESSION_IMPL *session, WT_CURSOR **cp, const char *pro
             } else
                 WT_RET(__pack_init(session, &pack, c->key_format));
             buf = &c->key;
-            p = (uint8_t *)buf->data;
-            end = p + buf->size;
+            end = p = (uint8_t *)buf->data;
+            if (end != NULL)
+                end += buf->size;
             continue;
```

Comment:
I wonder why we use `contunue` instead of `break` in this switch block? It looks unconventional to me.

Reply:
@AlexeyAnisimovAU, maybe I'm missing something, but `continue` and break will go to different places, that is, `break` does more processing in the current loop and `continue` does the next outer loop?

Reply:
Oh, my bad, sorry. I confused the scope of the parent `for` loop. I thought there's only one that `switch` block in the loop. `continue` makes sense now.